### DISCUSSION TITLE
Add some functions to make more customizable bondNote and atomNote

### DISCUSSION THREAD

**Discussion by WentongZhou** [rdkit/rdkit#7135] [Ideas]
Hi there,

"bondNote" and "atomNote" are great ways in rdkit to visualize some useful data attached to each specific atom or bond in one molecule. But it seems very limited to single font size and color. Is it possible to make more customizable bondNote and atomNote?
For example, can I use two different colors for bondNote and atomNote separately?
![image](https://github.com/rdkit/rdkit/assets/64748721/ddcae7fe-3c7e-4a90-9da5-241f6d4d51b8)

Thanks
Wentong

**Comment by WentongZhou** [comment]
It seems that  `bondNote` and `atomNote` don't support Unicode strings in Python.

**Comment by DavidACosgrove** [comment]
Hi @WentongZhou,
The atom and bond notes are intended to show small pieces of information about the bonds and angles.  You, the user, have no control over where they go.  If you want to add more than that, you can add your own text and shapes to the drawing wherever you like.  See this discussion and the associated gist from @greglandrum for an example. 
https://github.com/rdkit/rdkit/discussions/4832

**Reply by WentongZhou** [reply]
Hi David,

Thanks for your reply. I knew it was out of my control. I wonder if a color-changing option and support for Unicode strings could be added to show `bandNote` and `atomNote`. It will be handy. But if it is not easy to implement, using the method you mentioned to draw is okay.

Best,
Wentong

**Reply by greglandrum** [reply]
Unicode is a difficult one for us to support on the backend. We should do it eventually, but I don't think it's going to be easy.
